ctgan/data_transformer.py

```diff
@@ -36,43 +33,53 @@ def __init__(self, max_clusters=10, weight_threshold=0.005):
         self._max_clusters = max_clusters
         self._weight_threshold = weight_threshold
 
-    def _fit_continuous(self, column_name, raw_column_data):
-        """Train Bayesian GMM for continuous column."""
-        gm = BayesianGaussianMixture(
-            n_components=self._max_clusters,
-            weight_concentration_prior_type='dirichlet_process',
-            weight_concentration_prior=0.001,
-            n_init=1
-        )
+    def _fit_continuous(self, data):
+        """Train Bayesian GMM for continuous columns.
 
-        gm.fit(raw_column_data.reshape(-1, 1))
-        valid_component_indicator = gm.weights_ > self._weight_threshold
-        num_components = valid_component_indicator.sum()
+        Args:
+            data (pd.DataFrame):
+                A dataframe containing a column.
+
+        Returns:
+            namedtuple:
+                A ``ColumnTransformInfo`` object.
+        """
+        gm = BayesGMMTransformer()
+        gm.fit(data, list(data.columns))
+        num_components = sum(gm.valid_component_indicator)
```

Comment:
The `BayesGMMTransformer` filters the low weights out already right?

Reply:
Yes:
https://github.com/sdv-dev/RDT/blob/dbc550fc5e6cd62a032ab412aa8e2687bac82647/rdt/transformers/numerical.py#L600